CMP/Setup/SetupCMP/Helpers/InstallActionProcessor.cs

```diff
@@ -151,52 +151,48 @@ in PropertyBagDictionary.Instance.GetProperty<ArrayList>(PropertyBagConstants.It
+                bool isRpSuccess = true;
                 string RpFile = System.IO.Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location) + SetupConstants.RPFilename;
-                if (returnValue == SetupReturnValues.Successful && isUninstall != true && SetupConstants.WapDBOnRemoteServer == false)
+              
+                if (returnValue == SetupReturnValues.Successful && isUninstall != true )
                 {
-
-                    Process p = new Process();
-                    p.StartInfo.FileName = "Powershell.exe";
-                    p.StartInfo.Arguments = RpFile;
-                    p.StartInfo.UseShellExecute = false;
-                    p.StartInfo.RedirectStandardInput = true;
-                    p.StartInfo.RedirectStandardOutput = true;
-                    p.StartInfo.RedirectStandardError = true;
-                    p.StartInfo.CreateNoWindow = true;
-                    p.Start();
-                    errStr = p.StandardError.ReadToEnd();
-                    if (errStr != "") SetupLogger.LogError("Error in registering resource provider - " + errStr);
-                    p.Close();
-                    p.Dispose();
+                    WSManConnectionInfo connectionInfo = new WSManConnectionInfo();
+                    string computerName = "";
+                    if (SetupConstants.WapDBOnRemoteServer)
+                      computerName = SetupDatabaseHelper.GetAdminApiMachineName(true);
+                    else
+                      computerName=  Environment.MachineName;
+                    if (computerName != "")
+                    {
+                        connectionInfo.ComputerName = computerName;
+                        Runspace runspace = RunspaceFactory.CreateRunspace(connectionInfo);
+                        runspace.Open();
+                        StreamReader sr = new StreamReader(RpFile);
+                        string script = sr.ReadToEnd();
+                        using (PowerShell ps = PowerShell.Create())
+                        {
+                            ps.Runspace = runspace;
+                            ps.AddScript(script);
+                            var results = ps.Invoke();
+                            if (ps.Streams.Error.Count > 0)
+                            {
+                                isRpSuccess = false;
+                                SetupLogger.LogError("Error in registering resource provider - " + ps.Streams.Error.Count);
+                            }
+                            else
+                                SetupLogger.LogInfo("Registering resource provider is successful!!");
+                        }
+                        runspace.Close();
+                    }
+                    else
+                    {
+                        isRpSuccess = false;
+                        SetupLogger.LogError("Couldnt get Admin Api machine name and hence RP registatrion powershell script didnt run.");

```

Comment:
Minor grammar details in the "Couldn't", "Didn't", and "registration". :) We do need to correct them, though, just to make sure it looks polished.

Maybe we can rewrite the message to something like: "Could not resolve the server name for the Admin API. This has aborted the PowerShell script for the Azure Pack Connector Resource Provider registration."

@rotata In the case above, should the message give out a tip to run the RP registration script manually?